pinot-core/src/main/java/org/apache/pinot/core/data/manager/offline/DimensionTableDataManager.java

```diff
@@ -118,17 +120,20 @@ public void removeSegment(String segmentName) {
    */
   private void loadLookupTable()
       throws Exception {
-    Map<PrimaryKey, GenericRow> snapshot;
-    Map<PrimaryKey, GenericRow> replacement;
+//    Map<PrimaryKey, GenericRow> snapshot;
+//    Map<PrimaryKey, GenericRow> replacement;
+    DimensionTable snapshot;
+    DimensionTable replacement;
     do {
-      snapshot = _lookupTable;
-      replacement = new HashMap<>(snapshot.size());
+      snapshot = _dimensionTable;
+      replacement = new DimensionTable();
       populate(replacement);
```

Comment:
Could be `replacement = populate();` then the class's data doesn't need to be mutable.